### PR TITLE
Fix overwritng configjson during parsing subconfigs

### PR DESCRIPTION
**Pull request by michalkulakowski**

### CODE REVIEW

File: src/modelmanager.cpp [hunk 1]
```diff
@@ -791,20 +791,21 @@ Status ModelManager::loadModelsConfig(rapidjson::Document& configJson, std::vect
             SPDLOG_LOGGER_DEBUG(modelmanager_logger, "Subconfig path: {} provided for graph: {} does not exist. Loading subconfig models will be skipped.", subconfigPath, subconfigPath);
             continue;
         }
+        rapidjson::Document subconfigJson;
         rapidjson::IStreamWrapper isw(ifs);
-        rapidjson::ParseResult parseResult = configJson.ParseStream(isw);
+        rapidjson::ParseResult parseResult = subconfigJson.ParseStream(isw);
         if (!parseResult) {
             SPDLOG_LOGGER_ERROR(modelmanager_logger, "Mediapipe: {} configuration: {} file is not a valid JSON file. Error: {}",
```
Comment (mzegla) [inline src/modelmanager.cpp:798]:
```suggestion
            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Mediapipe graph subconfig: {} configuration: {} file is not a valid JSON file. Error: {}",
```

File: src/modelmanager.cpp [hunk 2]
```diff
@@ -791,20 +791,21 @@ Status ModelManager::loadModelsConfig(rapidjson::Document& configJson, std::vect
...
                 mediapipeConfig.getGraphName(), subconfigPath, rapidjson::GetParseError_En(parseResult.Code()));
             return StatusCode::JSON_INVALID;
         }
-        if (validateJsonAgainstSchema(configJson, MEDIAPIPE_SUBCONFIG_SCHEMA.c_str()) != StatusCode::OK) {
+        if (validateJsonAgainstSchema(subconfigJson, MEDIAPIPE_SUBCONFIG_SCHEMA.c_str()) != StatusCode::OK) {
             SPDLOG_LOGGER_ERROR(modelmanager_logger, "Mediapipe configuration file is not in valid configuration format");
```
Comment (mzegla) [inline src/modelmanager.cpp:803]:
```suggestion
            SPDLOG_LOGGER_ERROR(modelmanager_logger, "Mediapipe graph subconfig file is not in valid configuration format");
```

File: src/test/mediapipeflow_test.cpp [hunk 3]
```diff
@@ -564,6 +565,24 @@ TEST_F(MediapipeConfig, MediapipeAdd) {
     manager.join();
 }
 
+TEST_F(MediapipeConfig, MediapipeDummyWithDag) {
```
Comment (mzegla) [inline src/test/mediapipeflow_test.cpp:568]: Can we also test if the model from the graph subconfig has been loaded?

**Comment by atobiszei** [comment]
What was the issue here? Original content of main config.json was overridden by subconfig? If so then we should have a test that MP loads (and executes) correctly when there is no required model in subconfig.json, but the subconfig json exist and is empty (model_config_list:[])

**Review by atobiszei — APPROVED** [review]

**Review by mzegla — APPROVED** [review]

**Pull request merged**